skaha/src/main/java/org/opencadc/skaha/session/SessionAction.java

```diff
@@ -291,6 +283,33 @@ protected String getImageName(String image) {
 
     }
     
+    protected synchronized void injectFile(String data, String posixID, String userid) throws IOException, InterruptedException {
+        // stage file
+        String tmpFileName = "/tmp/" + UUID.randomUUID();
+        File file = new File(tmpFileName);
+        if (!file.setExecutable(true, true)) {
+            log.debug("Failed to set execution permssion on file " + tmpFileName);
+        }
+        BufferedWriter writer = new BufferedWriter(new FileWriter(file));
+        writer.write(data + "\n");
+        writer.flush();
+        writer.close();
+        
+        // update file permissions
+        String[] chown = new String[] {"chown", posixID + ":" + posixID, tmpFileName};
+        execute(chown);
+//        String[] chown = new String[] {"chown", "-R", "guest:guest", "/home/" + userid + "/.ssl"};
+//        execute(chown);
+        
+        // inject file
+        String[] injectCert = new String[] {"cp",  "-rp", tmpFileName, homedir + "/" + userid + "/.ssl/cadcproxy.pem"};
```

Comment:
@yeunga From what I've read (though I'm not an expert), the `cp` command is not atomic, but the `mv` command is...

Reply:
Sure, I can change from 'cp' to 'mv' and remove the clean up code.